pytorch_lightning/trainer/training_loop.py

```diff
@@ -507,7 +507,7 @@ def run_training_epoch(self):
     def check_checkpoint_callback(self, should_check_val):
         # when no val loop is present or fast-dev-run still need to call checkpoints
         # TODO bake this logic into the checkpoint callback
-        should_activate = not self.is_overridden('validation_step') and not (self.fast_dev_run or should_check_val)
```

Comment:
should check val covers the case for fast dev run, right?

Reply:
yes and yes.

Reply:
`should_check_val` is always True in all possible cases.

Reply:
The test I added just checks the batches in case of `fast_dev_run`. It failed with an error on master.

Reply:
yeah, should_check_val covers it.